dev-tools/oxlog/src/lib.rs

```diff
@@ -0,0 +1,448 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! A tool to show oxide related log file paths
+//!
+//! All data is based off of reading the filesystem
+
+use anyhow::Context;
+use camino::Utf8PathBuf;
+use chrono::{DateTime, Utc};
+use std::collections::BTreeMap;
+use std::fs::{read_dir, DirEntry};
```

Comment:
I believe camino supports a version of read_dir that does path parsing, and exposes a UTF-8 based DirEntry:
https://docs.rs/camino/1.1.6/camino/struct.Utf8Path.html#method.read_dir

That might be useful here?

Reply:
Great idea! This saves a 3 line check when getting the filename! Done in [5e48e48](https://github.com/oxidecomputer/omicron/pull/4810/commits/5e48e4814f0c0d5a6beebc81f123e0bd7d86faf8)